src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/CodecsTest.kt

```diff
@@ -129,11 +131,14 @@ class CodecsTest {
 
     @TestFactory
     fun dateTypesTests(): Collection<DynamicTest> {
-        val testData = listOf(
-            LocalDate.now(),
-            LocalTime.now(),
-            LocalDateTime.now(),
-            Instant.now(),
+        LocalTime.parse("16:36:38.035420").toString()
+        val testData = listOf<Pair<TemporalAccessor, (TemporalAccessor) -> String>>(
+            LocalDate.now() to TemporalAccessor::toString,
+            LocalTime.now() to DateTimeFormatter.ISO_LOCAL_TIME::format,
+            // Check case when LocalTime.toString() around nanos to 1000
+            LocalTime.parse("16:36:38.035420") to DateTimeFormatter.ISO_LOCAL_TIME::format,
+            LocalDateTime.now() to DateTimeFormatter.ISO_LOCAL_DATE_TIME::format,
+            Instant.now() to TemporalAccessor::toString,
         )
         return testData.map {
```

Comment:
You can use destruction declaration in `map` lambda for more readable code
```suggestion
        return testData.map { (value, formatter) ->
```